### PR TITLE
Blog - Managing Dependencies using Poetry in Python

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by balapriyac** [comment]
Thanks, Sameer. Will review this tomorrow morning.

File: blog/_posts/2023-05-08-python-poetry.md [hunk 1]
```diff
@@ -0,0 +1,712 @@
...
+
+You can use the [`new`](https://python-poetry.org/docs/cli/#options-1) command followed by the project name to create a new Poetry project. For example, you can use the following command to create a project named `weather-update` for your library:
+
+~~~{.bash caption=">_"}
+poetry new weather-update
+~~~
+
+The command creates a new folder `weather-update` with the following structure:
+
+~~~{ caption="file-hirarechy"}
```
Comment (balapriyac) [inline blog/_posts/2023-05-08-python-poetry.md:69]:
```suggestion
~~~{ caption=" "}
```

File: blog/_posts/2023-05-08-python-poetry.md [hunk 2]
```diff
@@ -0,0 +1,712 @@
...
+[tool.poetry.group.test.dependencies]
+pytest = "^7.2.1"
+requests-mock = "^1.10.0"
+~~~
+
+Non-optional groups in Poetry refer to the groups of dependencies that are required for a project to function properly. They are not designated as optional in the pyproject.toml file and their dependencies are considered part of the default set of dependencies for a project. Unlike optional groups, non-optional groups are always installed, and their dependencies are always available for use in your project.
+
+Optional and non-optional dependency groups are functionally independent and isolated. Packages in one group are not automatically made available to another group, and packages installed in one group do not affect packages installed in another group. This allows you to manage different sets of dependencies for different parts of your project, making it easier to manage the dependencies of your project as it evolves.
+
+> The default dependencies for a project in Poetry consist of two types of dependencies: the dependencies specified in the `tool.poetry.dependencies` table of the `pyproject.toml` file and the dependencies specified in the non-optional groups.
```
Comment (balapriyac) [inline blog/_posts/2023-05-08-python-poetry.md:522]:
```suggestion
The default dependencies for a project in Poetry consist of two types of dependencies: the dependencies specified in the `tool.poetry.dependencies` table of the `pyproject.toml` file and the dependencies specified in the non-optional groups.
```

**Review by balapriyac — CHANGES_REQUESTED** [review]
As always, looks great! Just requested a couple small changes.

File: blog/_posts/2023-05-08-python-poetry.md [hunk 3]
```diff
@@ -0,0 +1,712 @@
...
+The command will publish the package to the Python Package Index (PyPI), making them available for installation through Poetry.
+After publishing the package, you can search for it on PyPI. For example, you can find the `weather-update` library [PYPI](https://pypi.org/project/weather-update/). You can then install it on your system and try using it.
+
+## Conclusion
+
+Managing dependencies in Python development is crucial for ensuring the stability and [reliability](/blog/achieving-repeatability) of your projects. Poetry offers a convenient and efficient solution for handling dependencies, making it a popular choice among Python developers.
+
+In this tutorial, you have explored the steps involved in using Poetry for managing dependencies in Python projects, from installation to publishing a package. Specifically, you have learned the creation of virtual environments, managing dependencies, including adding, removing, and synchronizing them, and publishing a Python project as a package. You can explore much more about Poetry in its [official documentation](https://python-poetry.org/).
+
+{% include cta/cta1.html %}
```
Comment (balapriyac) [inline blog/_posts/2023-05-08-python-poetry.md:712]:
```suggestion
{% include cta/cta2.html %}
```

Comment (balapriyac) [reply blog/_posts/2023-05-08-python-poetry.md:712]: There's an enhanced CTA, cta2. Let's use that going forward. :)

**Review by balapriyac — CHANGES_REQUESTED** [review]

**Comment by msali123** [comment]
https://6426e4002bba3e25582c0adf--ephemeral-narwhal-3ac3b2.netlify.app/blog/python-poetry/
Changes done

File: blog/_posts/2023-05-08-python-poetry.md (hunk 3 above)
Comment (msali123) [reply blog/_posts/2023-05-08-python-poetry.md]: @balapriyac this isnt working

**Comment by balapriyac** [comment]
Oh, okay, Sameer. Please ignore that suggestion. I saw {% include_html
cta/cta2.html %} in the updated posts. So I thought it'd work.

File: blog/_posts/2023-05-08-python-poetry.md (hunk 3 above)
Comment (msali123) [reply blog/_posts/2023-05-08-python-poetry.md]: image

**Comment by balapriyac** [comment]
@jalletto The updated CTA isn't appearing. Besides that, the article is ready to merge.

> https://6426e4002bba3e25582c0adf--ephemeral-narwhal-3ac3b2.netlify.app/blog/python-poetry/ Changes done

Hi Sameer, can you please add the most recent preview?

**Comment by msali123** [comment]
https://6426e4002bba3e25582c0adf--ephemeral-narwhal-3ac3b2.netlify.app/blog/python-poetry/


This is the most recent preview @balapriyac

**Comment by msali123** [comment]
https://6426fc4c4aa11702f4b1dc19--ephemeral-narwhal-3ac3b2.netlify.app/blog/python-poetry/

@balapriyac Preview the old CTA

File: blog/_posts/2023-05-08-python-poetry.md [hunk 4]
```diff
@@ -0,0 +1,712 @@
...
```
Comment (jalletto) [inline blog/_posts/2023-05-08-python-poetry.md:712]:
```suggestion
{% include_html cta/cta1.html %}
```

Comment (jalletto) [reply blog/_posts/2023-05-08-python-poetry.md:712]: I don't know the reason for this exactly, but now you need to do include_html for this to work.

**Review by jalletto — CHANGES_REQUESTED** [review]

**Comment by msali123** [comment]
@balapriyac @jalletto updated Preview [Link](https://64277437a714791625aeef3f--ephemeral-narwhal-3ac3b2.netlify.app/blog/python-poetry/) with {% include_html cta/cta2.html %}

File: blog/_posts/2023-05-08-python-poetry.md [hunk 5]
```diff
@@ -0,0 +1,712 @@
...
+        print("Error: You do not have permission to write to the file.")
+    except configparser.Error:
+        print("Error: Unable to write to the configuration file.")
+~~~
+
+The `set_api_key` function sets the API key in the configuration file using the `configparser` library. The function prompts the user to enter the API key and writes it to the file.
+
+To check whether the code works, you can create a `main.py` file and add the following code:
+
+~~~{.python caption="weather.py"}
```
Comment (balapriyac) [inline blog/_posts/2023-05-08-python-poetry.md:373]:
```suggestion
~~~{.python caption="main.py"}
```

**Review by balapriyac — CHANGES_REQUESTED** [review]
@msali123 there's one code block caption that I missed in the previous round of review.

**Comment by msali123** [comment]
Done @balapriyac 
[Link](https://642a9632ab8bc450333e3e8e--ephemeral-narwhal-3ac3b2.netlify.app/blog/python-poetry/)

**Comment by balapriyac** [comment]
@jalletto this is ready to merge.

**Review by jalletto — APPROVED** [review]

**Pull request merged**